src/Simplic.Flow.Editor.UI/ViewModel/ToolboxViewModel.cs

```diff
@@ -62,41 +80,148 @@ private void LoadGalleryItems()
                         {
-                            Name = item.Name,                            
+                            Name = item.Name,
                             DataContext = item,
                         };
                     }
 
-                    gallery.Items.Add(galleryItem);                    
+                    gallery.Items.Add(galleryItem);
                 }
-
                 GalleryItems.Add(gallery);
-            }            
-            
+                GalleryViewSources[gallery] = CollectionViewSource.GetDefaultView(gallery.Items);
+                GalleryViewSources[gallery].Filter = nodeFilter;
+            }
+
             SelectedGallery = GalleryItems.FirstOrDefault();
         }
-        #endregion
 
-        #endregion
-        
-        #region Public Properties
+        /// <summary>
+        /// Filters for relevant galleries.
+        /// </summary>
+        /// <param name="term">Term by which is filtered</param>
+        private bool galleryFilter(object obj)
+        {
+            if (obj is Gallery category)
+            {
+                bool hasRelevantNode = false;
+                foreach (var node in category.Items)
+                    if (nodeFilter(node))
+                        hasRelevantNode = true;
+                return hasRelevantNode;
+            }
+            return true;
+        }
 
-        #region [GalleryItems]
-        public ObservableCollection<Gallery> GalleryItems { get; set; }
-        #endregion
+        /// <summary>
+        /// Filters for relevant nodes.
+        /// </summary>
+        /// <param name="obj"></param>
+        /// <returns></returns>
+        private bool nodeFilter(object obj)
+        {
+            if (!(obj is GalleryItem))
+                return true;
 
-        #region [NodeDefinitions]
-        public IList<Definition.NodeDefinition> NodeDefinitions { get; }
-        #endregion
+            if (normalizeString(searchTerm) == string.Empty)
+                return true;
+
+            var node = obj as GalleryItem;
+
+            if (MatchWholeWord)
+            {
+                if (MatchCase)
+                    return node.Header.Split(' ').Contains(normalizeString(SearchTerm));
+
+                return node.Header.ToLower().Split(' ').Contains(normalizeString(SearchTerm).ToLower());
+            }
 
-        #region [SelectedGallery]
-        public Gallery SelectedGallery
+            if (MatchCase)
+                return node.Header.Contains(normalizeString(SearchTerm));
+
+            return node.Header.ToLower().Contains(normalizeString(SearchTerm).ToLower());
+        }
+
+        /// <summary>
+        /// Removes whitespaces from given string.
+        /// </summary>
+        /// <param name="str">string</param>
+        /// <returns>normalized string</returns>
+        private string normalizeString(string str)
+        {
+            return Regex.Replace(str, @"\s+", "");
+        }
```

Comment:
This might cause issues. When I search for e.g "Telematic Message"
and want to find "Send Telematic Massage" 
then the header does not contain TelematicMessage. Or am I wrong?

Maybe just stripping whitespaces at the beginning or end might help